src/main/java/org/broadinstitute/hellbender/metrics/InsertSizeMetricsCollector.java

```diff
@@ -1,50 +1,66 @@
+    private static final Logger log = LogManager.getLogger(InsertSizeMetricsCollector.class);
 
-    // When generating the Histogram, discard any data categories (out of FR, TANDEM, RF) that have fewer than this
-    // percentage of overall reads. (Range: 0 to 1)
-    private final double minimumPct;
-
-    // Generate mean, sd and plots by trimming the data down to MEDIAN + DEVIATIONS*MEDIAN_ABSOLUTE_DEVIATION.
-    // This is done because insert size data typically includes enough anomolous values from chimeras and other
-    // artifacts to make the mean and sd grossly misleading regarding the real distribution.
-    private final double deviations;
-
-    //Explicitly sets the Histogram width, overriding automatic truncation of Histogram tail.
-    //Also, when calculating mean and stdev, only bins <= HISTOGRAM_WIDTH will be included.
-    private Integer HistogramWidth;
-
-    public InsertSizeMetricsCollector(final Set<MetricAccumulationLevel> accumulationLevels, final List<SAMReadGroupRecord> samRgRecords,
-                                      final double minimumPct, final Integer HistogramWidth, final double deviations) {
-        this.minimumPct = minimumPct;
-        this.HistogramWidth = HistogramWidth;
-        this.deviations = deviations;
-        setup(accumulationLevels, samRgRecords);
+    InsertSizeMetricsArgumentCollection inputArgs = null;
+
+    public InsertSizeMetricsCollector(
+            final InsertSizeMetricsArgumentCollection inputArgs,
+            final SAMFileHeader samHeader)
+    {
+        super();
+        this.inputArgs = inputArgs;
+        setup(inputArgs.metricAccumulationLevel, samHeader.getReadGroups());
     }
 
-    // We will pass insertSize and PairOrientation with the DefaultPerRecordCollectorArgs passed to the record collectors
-    // This method is called once Per samRecord
+    /**
+     * Return the read filter for InsertSizeMetrics collector.
+     * @param samHeader SAMFileHeader for the input file
+     * @return ReadFilter to be used to filter records
+     */
+    public ReadFilter getReadFilter(SAMFileHeader samHeader) {
+        return new WellformedReadFilter(samHeader).and(
+                new InsertSizeMetricsReadFilter(
+                    inputArgs.useEnd,
+                    inputArgs.filterNonProperlyPairedReads,
+                    !inputArgs.useDuplicateReads,
+                    !inputArgs.useSecondaryAlignments,
+                    !inputArgs.useSupplementaryAlignments,
+                    inputArgs.MQPassingThreshold
+            )
+        );
+    }
+
+    // We will pass insertSize and PairOrientation with the DefaultPerRecordCollectorArgs passed to
+    // the record collectors. This method is called once per samRecord
     @Override
     protected InsertSizeMetricsCollectorArgs makeArg(SAMRecord samRecord, ReferenceSequence refSeq) {
         final int insertSize = Math.abs(samRecord.getInferredInsertSize());
```

Comment:
I'm not sure why its this way - this code is in the existing walker implementation. The Picard code also does the same thing.